llama_index/node_parser/interface.py

```diff
@@ -22,3 +22,18 @@ def get_nodes_from_documents(
             documents (Sequence[Document]): documents to parse
 
         """
+
+
+class NodeParserPostProcessor(ABC):
+    """Base interface for node parser post processor."""
+
+    @abstractmethod
+    def post_process_nodes(
```

Comment:
not a huge fan of this name (and the class name), would prefer the class to be `BaseMetadataExtractor` and the function to be `extract_metadata` 

also in general to make this a tighter function signature, i wonder if it would be nicer if this was stateless - e.g. return a Dict in metadata. the outer class (the current class `MetadataAugmentationPostProcessor`) can actually decide to 1) merge the metadata from different fields, and 2) set the metadata on the nodes

Reply:
I agree here. Feels a little nicer to work with when nodes in, metadata dict out

Reply:
Hmm, sounds good.